fdbserver/SimulatedCluster.actor.cpp

```diff
@@ -645,6 +645,283 @@ T simulate(const T& in) {
 	return out;
 }
 
+ACTOR Future<std::string> getLayerStatus(Reference<ReadYourWritesTransaction> tr,
+                                         IPAddress localIP,
+                                         std::string name,
+                                         std::string id,
+                                         Database dest,
+                                         Snapshot snapshot = Snapshot::False) {
+	// This process will write a document that looks like this:
+	// { backup : { $expires : {<subdoc>}, version: <version from approximately 30 seconds from now> }
+	// so that the value under 'backup' will eventually expire to null and thus be ignored by
+	// readers of status.  This is because if all agents die then they can no longer clean up old
+	// status docs from other dead agents.
+
+	state Version readVer = wait(tr->getReadVersion());
+
+	state json_spirit::mValue layersRootValue; // Will contain stuff that goes into the doc at the layers status root
+	JSONDoc layersRoot(layersRootValue); // Convenient mutator / accessor for the layers root
+	JSONDoc op = layersRoot.subDoc(name); // Operator object for the $expires operation
+	// Create the $expires key which is where the rest of the status output will go
+
+	state JSONDoc layerRoot = op.subDoc("$expires");
+	// Set the version argument in the $expires operator object.
+	op.create("version") = readVer + 120 * CLIENT_KNOBS->CORE_VERSIONSPERSECOND;
+
+	layerRoot.create("instances_running.$sum") = 1;
+	layerRoot.create("last_updated.$max") = now();
+
+	state JSONDoc o = layerRoot.subDoc("instances." + id);
+
+	o.create("version") = FDB_VT_VERSION;
+	o.create("id") = id;
+	o.create("last_updated") = now();
+	// o.create("memory_usage") = (int64_t)getMemoryUsage();
+	// o.create("resident_size") = (int64_t)getResidentMemoryUsage();
+	// o.create("main_thread_cpu_seconds") = getProcessorTimeThread();
+	// o.create("process_cpu_seconds") = getProcessorTimeProcess();
+	// o.create("configured_workers") = CLIENT_KNOBS->BACKUP_TASKS_PER_AGENT;
+	// o.create("processID") = ::getpid();
+	// o.create("locality") = tr->getDatabase()->clientLocality.toJSON<json_spirit::mObject>();
+	// o.create("networkAddress") = localIP.toString();
```

Comment:
Remove this commented out stuff? Or its for debugging?

Reply:
This PR is for debugging.